tests/functional/test_timeformat.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2014 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Change the scalar response parsing behavior for the AWS CLI.
+
+The underlying library used by botocore has some response parsing
+behavior that we'd like to modify in the AWS CLI.  There are two:
+
+    * Parsing binary content.
+    * Parsing timestamps (dates)
+
+For the first option we can't print binary content to the terminal,
+so this customization leaves the binary content base64 encoded.  If the
+user wants the binary content, they can then base64 decode the appropriate
+fields as needed.
+
+There's nothing currently done for timestamps, but this will change
+in the future.
+
+"""
+import os
+import mock
+import datetime
+from dateutil.tz import tzlocal
+from awscli.testutils import unittest, FileCreator
+from awscli.clidriver import create_clidriver
+
+
+class TestParsers(unittest.TestCase):
+    def setUp(self):
+        self.environ = {
+            'AWS_DATA_PATH': os.environ['AWS_DATA_PATH'],
+            'AWS_DEFAULT_REGION': 'us-east-1',
+            'AWS_ACCESS_KEY_ID': 'access_key',
+            'AWS_SECRET_ACCESS_KEY': 'secret_key',
+            'AWS_CONFIG_FILE': '',
+            'AWS_SHARED_CREDENTIALS_FILE': '',
+        }
+        self.environ_patch = mock.patch('os.environ', self.environ)
+        self.environ_patch.start()
```

Comment:
Note for any other reviewers: ``BaseAWSCommandParamsTest`` is not being used because we need to mock out the wire response and patching ``make_request`` on endpoint also patches out the response parsing.